jerry-core/ecma/builtin-objects/ecma-builtin-helpers-date.cpp

```diff
@@ -741,17 +741,20 @@ ecma_date_make_day (ecma_number_t year, /**< year value */
   mn = (mn < 0) ? 12 + mn : mn;
   ecma_number_t time = ecma_date_time_from_year (ym);
 
-  JERRY_ASSERT (ecma_date_year_from_time (time) == ym);
-
-  while (ecma_date_month_from_time (time) < mn)
+  if (ecma_date_year_from_time (time) == ym)
   {
-    time += ECMA_DATE_MS_PER_DAY;
-  }
+    while (ecma_date_month_from_time (time) < mn)
+    {
+      time += ECMA_DATE_MS_PER_DAY;
```

Comment:
Could we remove the the loop by calculating necessary difference and adding it at once, so improving performance?

If not, we could:
- add 28 days (minimal number of days in a month) at first step, as initial day number is 1, so reducing number of iterations to at most 3;
- or use binary search for calculating `time` value.